File: src/skills/getConversationHistory.js
import { query } from '../db/pool.js';

const MAX_MESSAGES = 5;

/**
 * Retrieve the last N messages for a given conversation (short-term memory).
 *
 * @param {string} conversationId
 * @param {number} [limit=5]  Number of recent messages to return
 * @returns {Promise<Array<{role: string, content: string, metadata: object, createdAt: string}>>}
 */
export default async function getConversationHistory(conversationId, limit = MAX_MESSAGES) {
  if (!conversationId) {
    return [];
  }

  try {
    const result = await query(
      `SELECT role, content, metadata, created_at AS "createdAt"
       FROM messages
       WHERE conversation_id = $1
       ORDER BY created_at DESC
       LIMIT $2`,
      [conversationId, limit],
    );

    // Return in chronological order (oldest first)
    return result.rows.reverse();
  } catch (err) {
    console.error('[getConversationHistory] Failed to fetch history:', err.message);
    return [];
  }
}

File: app/skills/get_conversation_history.py
"""
skills/get_conversation_history.py — Short-term memory: last N messages.
"""

from app.db.pool import query

MAX_MESSAGES = 5


def get_conversation_history(conversation_id: str, limit: int = MAX_MESSAGES) -> list[dict]:
    """Return last `limit` messages in chronological order (oldest first)."""
    if not conversation_id:
        return []

    try:
        rows = query(
            """
            SELECT role, content, metadata, created_at AS "created_at"
            FROM messages
            WHERE conversation_id = %s
            ORDER BY created_at DESC
            LIMIT %s
            """,
            (conversation_id, limit),
        )
        # Reverse to get chronological order
        return list(reversed(rows))
    except Exception as err:
        print(f"[get_conversation_history] Failed: {err}")
        return []
